### PR TITLE
Add PowerShell SMS example using Twillo

### LINKED ISSUE #582: Add an SMS function example using Twillio
**Is your feature request related to a problem? Please describe.**
Demonstrate SMS (text message) function using Twillio

### PR DESCRIPTION
**Pull request by lamw**
Closes #582 
Signed-off-by: William Lam <wlam@vmware.com>

## Summary

SMS message sent via Twillo when a VM snapshot is created 

screenshot-1

## Pull Request Checklist
🚨 Please review the [guidelines for contributing](https://vmweventbroker.io/community) to this repository.

- [ ] Please ensure that you are making a pull request against the **Development** branch
- [ ] Please use the `WIP` keyword in the title of your PR if you are not ready for review
- [ ] Please ensure that you have opened a Github Issue if you are resolving/fixing a problem
- [ ] Please ensure that you have [signed](https://help.github.com/en/github/authenticating-to-github/signing-commits) all commits and that you have [squashed](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) all relevant commits related to your change
- [ ] Please make sure that you have tested your change locally by successfully [building and deploying the VMware Event Broker Appliance](https://vmweventbroker.io/kb/contribute-appliance) and/or [building and deploying VMware Event Router](https://vmweventbroker.io/kb/contribute-eventrouter)
- [ ] Please include any relevant screenshots and/or output as part of your testing
- [ ] Please include any documentation updates that is applicable for your changes

## Change Type

What types of changes does your code introduce to the VMware Event Broker Appliance?

_Put an `x` in all boxes that apply_

Please check the type of change your PR introduces:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation changes
- [ ] Other (please describe):

### CODE REVIEW

File: examples/knative/powershell/kn-ps-twillio-sms/README.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+`default` Rabbit `broker`. The Knative `service` and `trigger` will be installed in the
+`vmware-functions` Kubernetes namespace, assuming that the `broker` is also available there.
+
+Push your container image to an accessible registry such as Docker once you're done developing and testing your function logic.
+
+```console
+docker push <docker-username>/kn-ps-twillo-sms:${TAG}
+```
+
+Update the `sms_secret.json` file with you Twillo credentials and configurations and then create the kubernetes secret which can then be accessed from within the function by using the environment variable named called `SMS_SECRET`.
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-twillio-sms/README.md:88]:
```suggestion
Update the `sms_secret.json` file with your Twillo credentials and configuration and then create the kubernetes secret which can then be accessed from within the function by using the environment variable named `SMS_SECRET`.
```

File: examples/knative/powershell/kn-ps-twillio-sms/README.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+```console
+# create secret
+
+kubectl -n vmware-functions create secret generic sms-secret --from-file=SMS_SECRET=sms_secret.json
+
+# update label for secret to show up in VEBA UI
+kubectl -n vmware-functions label secret sms-secret app=veba-ui
+```
+
+Edit the `function.yaml` file with the name of the container image from Step 1 if you made any changes. If not, the default VMware container image will suffice. By default, the function deployment will filter on the `TaskEvent` vCenter Server Event and VM snapshot creation task for more details on identifying a specific task descriptionId from a TaskEvent, please [see this blog post](https://williamlam.com/2019/02/creating-vcenter-alarms-based-on-task-events-such-as-folder-creation.html) for more details. If you wish to change this, update the `subject` field within `function.yaml` to the desired event type.
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-twillio-sms/README.md:99]:
```suggestion
Edit the `function.yaml` file with the name of the container image from Step 1 if you made any changes. If not, the default VMware container image will suffice. By default, the function deployment will filter on the `TaskEvent` vCenter Server Event and VM snapshot creation task. For more details on identifying a specific task descriptionId from a TaskEvent, please [see this blog post](https://williamlam.com/2019/02/creating-vcenter-alarms-based-on-task-events-such-as-folder-creation.html) for more details. If you wish to change this, update the `subject` field within `function.yaml` to the desired event type.
```

File: examples/knative/powershell/kn-ps-twillio-sms/README.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-twillio-sms/README.md:100]: Remove empty line

File: examples/knative/powershell/kn-ps-twillio-sms/README.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+Deploy the function to the VMware Event Broker Appliance (VEBA).
+
+```console
+# deploy function
+
+kubectl -n vmware-functions apply -f function.yaml
+```
+
+For testing purposes, the `function.yaml` contains the following annotations, which will ensure the Knative Service Pod will always run **exactly** one instance for debugging purposes. Functions deployed through through the VMware Event Broker Appliance UI defaults to scale to 0, which means the pods will only run when it is triggered by an vCenter Event.
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-twillio-sms/README.md:110]:
```suggestion
For testing purposes, the `function.yaml` contains the following annotations, which will ensure the Knative Service Pod will always run **exactly** one instance for debugging purposes. Functions deployed through the VMware Event Broker Appliance UI defaults to scale to 0, which means the pods will only run when it is triggered by an vCenter Event.
```

File: examples/knative/powershell/kn-ps-twillio-sms/function.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+kind: Trigger
+metadata:
+  name: kn-ps-twillo-sms-trigger
+  labels:
+    app: veba-ui
+spec:
+  broker: default
+  filter:
+    attributes:
+      type: com.vmware.event.router/event
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-twillio-sms/function.yaml:33]: Feel free to drop

File: examples/knative/powershell/kn-ps-twillio-sms/handler.ps1 [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+   try {
+      $jsonSecrets = ${env:SMS_SECRET} | ConvertFrom-Json
+   } catch {
+      throw "`nK8s secrets `$env:SMS_SECRET does not look to be defined"
+   }
+
+   # Check TaskEvent to make sure its VM Create Snapshot Event
+   # See https://williamlam.com/2019/02/creating-vcenter-alarms-based-on-task-events-such-as-folder-creation.html for more details on finding descriptionId from TaskEvent
+   if($cloudEventData.Info.DescriptionId -eq "VirtualMachine.createSnapshot") {
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-twillio-sms/handler.ps1:38]: Perhaps an else or default clause would be neat to log that it skipped this TaskEvent.

File: examples/knative/powershell/kn-ps-twillio-sms/handler.ps1 [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+      # Make API request, selecting JSON properties from response
+      Write-Host "$(Get-Date) - Sending SMS Message"
+      try {
+         $response = Invoke-WebRequest $url -Method Post -Credential $credential -Body $params
+         $StatusCode = $Response.StatusCode
+      } catch {
+         throw "$(Get-Date) - Failed to send SMS: $($_)"
+      }
+
+      if($StatusCode -eq "201") {
```
Comment (embano1) [inline examples/knative/powershell/kn-ps-twillio-sms/handler.ps1:81]: Else clause with other status code would be neat to log

**Review by embano1 — CHANGES_REQUESTED** [review]
Minor changes requested

**Comment by lamw** [comment]
All feedback addressed

**Review by embano1 — APPROVED** [review]
LGTM

**Review by embano1 — APPROVED** [review]
LGTM

**Pull request merged**